sdk/messaging/azeventhubs/internal/exported/connection_string_properties.go

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package exported
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+// ConnectionStringProperties are the properties of a connection string
+// as returned by [NewConnectionStringProperties].
+type ConnectionStringProperties struct {
+	// Endpoint is the Endpoint value in the connection string.
+	// Ex: sb://example.servicebus.windows.net
+	Endpoint *string
+
+	// EntityPath is EntityPath value in the connection string.
+	EntityPath *string
+
+	// FullyQualifiedNamespace is the Endpoint value without the protocol scheme.
+	// Ex: example.servicebus.windows.net
+	FullyQualifiedNamespace *string
```

Comment:
If this, and others, are always required then IMO they shouldn't be pointer-to-type.

Reply:
Done. FullyQualifiedNamespace and Endpoint are the only ones that are guaranteed to be there - the others are mutually exclusive so we'll leave those as pointerse.